src/main/java/seedu/address/commons/util/CsvFile.java

```diff
@@ -0,0 +1,197 @@
+package seedu.address.commons.util;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import seedu.address.commons.exceptions.CsvMismatchedColumnException;
+import seedu.address.commons.exceptions.CsvMissingFieldException;
+
+/**
+ * Representation of the contents of a CSV file as an object.
+ */
+public class CsvFile {
+    private final int numColumns;
+
+    private final String delimiter;
+    private final HashMap<String, Integer> columnIndices;
+    private final String header;
+    private final List<CsvRow> rows;
+
+    /**
+     * Represents a row of values in the CSV file. Encapsulating the row values
+     * within this object allows one to extract the value in a column with just
+     * the column's name, without having to know the column index.
+     */
+    public class CsvRow implements GetValuer {
```

Comment:
Any reason why the CSVRow class is not a separate class but rather a nested class within the CsvFile class?

Reply:
The CsvRow is designed to allow queries for values using the column header names so that the user does not need to know the ordering of the headers, which could easily change (e.g. shifting columns around in Excel).

For this feature to work, the header names must be mapped to the column indices, fulfilled with the columnIndices hashmap in the CsvFile class. This mapping is located in the CsvFile class to avoid having each CsvRow contain its copy of the mapping, which seems less space-efficient. However, due to this design, a CsvRow cannot exist in isolation from the CsvFile it belongs to. Hence, it is a nested class within the CsvFile class.